src/daos/wait_for.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+# parameter 1: number of seconds the timeout waits
+# parameter 2: function that returns 0 when the condition has been satisfied
+# parameter 3: message to be displayed upon timeout
+function wait_for()
+{
+    local timeout_limit=$1
+    local counter=0
+    $2
+    while [[ ! $? == 0 ]]; do
+        if [ $counter -ge $timeout_limit ]; then
+            echo "Timeout: $3"
+            exit 1
+        fi
+        sleep 1
+        $2
+        ((counter++))
```

Comment:
"$2" should be the last statement of the while loop to get accurate exit status [$?] in while condition.

Reply:
But let's not fix it in this PR. Let's first land this and then we can improve later. Its already a huge change made by someone else.